### PR TITLE
Feature/pytorch backend

### PR DESCRIPTION
**Pull request by MichaelMarien**
First attempt to add PyTorch backend, expect a lot of comments & input on this PR and there is no hurry as I might not have time to address those quickly. Consider this a first check of progress.

### CODE REVIEW

File: tensornetwork/tensornetwork_test.py [hunk 1]
```diff
@@ -1165,6 +1165,8 @@ def test_switch_backend():
 
 
 def test_svd_consistency(backend):
+  if backend == "pytorch":
+    pytest.skip()
```
Comment (Thenerdstation) [inline tensornetwork/tensornetwork_test.py]: Be sure to add "PyTorch doesn't support complex numbers" as the skip reason.

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 numpy>=1.16
 tensorflow>=1.13
+torch>=1.1
```
Comment (Thenerdstation) [inline requirements.txt]: We prefer to not force the user to install Pytorch if they don't need it. Can you put this line in the `requirements_travis.txt` file instead?

File: pylintrc [hunk 3]
```diff
@@ -160,7 +160,7 @@ additional-builtins=
 ignore-mixin-members=yes
 # List of classes names for which member attributes should not be checked
 # (useful for classes with attributes dynamically set).
-ignored-classes=SQLObject,twisted.internet.reactor,hashlib,google.appengine.api.memcache
+ignored-classes=SQLObject,twisted.internet.reactor,hashlib,google.appengine.api.memcache,torch
```
Comment (Thenerdstation) [inline pylintrc]: Any reason you added this?

File: tensornetwork/tensornetwork_test.py (hunk 1 above)
Comment (MichaelMarien) [reply tensornetwork/tensornetwork_test.py]: Done!

File: requirements.txt (hunk 2 above)
Comment (MichaelMarien) [reply requirements.txt]: makes a lot of sense

File: pylintrc (hunk 3 above)
Comment (MichaelMarien) [reply pylintrc]: It is to fix this issue: https://github.com/pytorch/pytorch/issues/701
I tried adding the separate functions that are used only, but for some reason that didn't work, so I reverted to the proposed solution of adding the full toch library. Tbh, I'm not 100% happy with this either.

**Comment by Thenerdstation** [comment]
https://github.com/google/TensorNetwork/issues/53

**Pull request merged**